rumqttc/src/tls.rs

```diff
@@ -1,34 +1,59 @@
 use tokio::net::TcpStream;
+
+#[cfg(feature = "use-rustls")]
 use tokio_rustls::rustls;
+#[cfg(feature = "use-rustls")]
 use tokio_rustls::rustls::client::InvalidDnsNameError;
+#[cfg(feature = "use-rustls")]
 use tokio_rustls::rustls::{
     Certificate, ClientConfig, OwnedTrustAnchor, PrivateKey, RootCertStore, ServerName,
 };
+#[cfg(feature = "use-rustls")]
 use tokio_rustls::webpki;
-use tokio_rustls::{client::TlsStream, TlsConnector};
-
-use crate::{Key, MqttOptions, TlsConfiguration};
+#[cfg(feature = "use-rustls")]
+use tokio_rustls::TlsConnector as RustlsConnector;
 
+#[cfg(feature = "use-rustls")]
+use crate::Key;
+#[cfg(feature = "use-rustls")]
 use std::convert::TryFrom;
-use std::io;
+#[cfg(feature = "use-rustls")]
 use std::io::{BufReader, Cursor};
-use std::net::AddrParseError;
+#[cfg(feature = "use-rustls")]
 use std::sync::Arc;
 
+use crate::framed::N;
+use crate::{MqttOptions, TlsConfiguration};
+
+#[cfg(feature = "use-native-tls")]
+use tokio_native_tls::TlsConnector as NativeTlsConnector;
+
+#[cfg(feature = "use-native-tls")]
+use tokio_native_tls::native_tls::Error as NativeTlsError;
+
+use std::net::AddrParseError;
+use std::io;
+
 #[derive(Debug, thiserror::Error)]
 pub enum Error {
     #[error("Addr")]
     Addr(#[from] AddrParseError),
     #[error("I/O")]
     Io(#[from] io::Error),
+    #[cfg(feature = "use-rustls")]
     #[error("Web Pki")]
     WebPki(#[from] webpki::Error),
+    #[cfg(feature = "use-rustls")]
     #[error("DNS name")]
     DNSName(#[from] InvalidDnsNameError),
+    #[cfg(feature = "use-rustls")]
     #[error("TLS error")]
     TLS(#[from] rustls::Error),
```

Comment:
We could name this Rustls(...)

Reply:
I would be in favor of that, but I wanted to keep the features consistent with rumqttd. I'll leave to your judgement whether you want to change the names just here, in both or in neither.